manifests/tftp.pp

```diff
@@ -1,50 +1,64 @@
 # Set up the tftp component
-class foreman_proxy::tftp {
+class foreman_proxy::tftp (
+  $user = $::foreman_proxy::user,
+  $root = $::foreman_proxy::tftp_root,
+  $directories = $::foreman_proxy::tftp_dirs,
+  $syslinux_filenames = $::foreman_proxy::tftp_syslinux_filenames,
+  $manage_wget = $::foreman_proxy::tftp_manage_wget,
+  $wget_version = $::foreman_proxy::ensure_packages_version,
+) {
   class { '::tftp':
-    root => $foreman_proxy::tftp_root,
+    root => $root,
   }
 
-  file { $foreman_proxy::tftp_dirs:
+  file { $directories:
     ensure  => directory,
-    owner   => $foreman_proxy::user,
+    owner   => $user,
     mode    => '0644',
     require => Class['foreman_proxy::install', 'tftp::install'],
     recurse => true,
   }
 
-  file { "${foreman_proxy::tftp_root}/grub2/grub.cfg":
+  file { "${root}/grub2/grub.cfg":
     ensure  => file,
-    owner   => $foreman_proxy::user,
+    owner   => $user,
     mode    => '0644',
-    content => template('foreman_proxy/grub.cfg.erb'),
+    content => file('foreman_proxy/grub.cfg'),
   }
 
-  foreman_proxy::tftp::copy_file { $foreman_proxy::tftp_syslinux_filenames:
-    target_path => $foreman_proxy::tftp_root,
-    require     => Class['foreman_proxy::install', 'tftp::install'],
+  $syslinux_filenames.each |$source_file| {
+    $filename = basename($source_file)
+    file {"${root}/${filename}":
+      ensure  => file,
+      source  => $source_file,
+      require => Class['foreman_proxy::install', 'tftp::install'],
+    }
   }
 
-  if $foreman_proxy::tftp_manage_wget {
-    ensure_packages(['wget'], { ensure => $foreman_proxy::ensure_packages_version, })
+  if $manage_wget {
+    ensure_packages(['wget'], { ensure => $wget_version, })
   }
 
   case $::osfamily {
     'RedHat': {
-      $osreleasemajor = regsubst($::operatingsystemrelease, '^(\d+)\..*$', '\1') # workaround for the possibly missing operatingsystemmajrelease
-      if versioncmp($osreleasemajor, '6') <= 0 {
+      if versioncmp($::operatingsystemmajrelease, '6') <= 0 {
```

Comment:
Nitpick, you can rely on `$facts['operatingsystemmajrelease']` or even `$facts['os']['release']['major']` if you want to completely modernise this for Puppet 4.